src/averageAges.js

```diff
@@ -1,61 +1,66 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const callBackFilter = (person) =>
+    century > 0
+      ? (person.sex === 'm' && Math.ceil(person.died / 100) === century)
+      : person.sex === 'm';
+
+  const callBackReduce = (sum, person) => sum + (person.died - person.born);
+  const mans = people.filter(callBackFilter);
+  const result = mans.reduce(callBackReduce, 0) / mans.length;
+
+  return result;
 }
 
 /**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
  * @param {object[]} people
  * @param {boolean} withChildren - optional
  *
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const callBackFilter = (person) =>
+    withChildren === true
```

Comment:
`withChildren` is boolean, so comparing with `=== true` is unnecessary. Probably, this variable better to rename in `hasChildren` (to show that it's bool), but it's naming given by task...